src/Integration/Connection/ConnectionWorkflow.cs

```diff
@@ -114,76 +120,91 @@ private void OnProjectsChanged(ConnectionInformation connection, ProjectInformat
             this.host.ActiveSection?.UserNotifications?.HideNotification(NotificationIds.FailedToConnectId);
-            this.host.ActiveSection?.UserNotifications?.HideNotification(NotificationIds.BadServerPluginId);
+            this.host.ActiveSection?.UserNotifications?.HideNotification(NotificationIds.BadSonarQubePluginId);
 
             notifications.ProgressChanged(connection.ServerUri.ToString());
 
-            notifications.ProgressChanged(Strings.ConnectionStepValidatinCredentials);
-            if (!this.host.SonarQubeService.AreCredentialsValid(connection, cancellationToken))
+            try
             {
-                AbortWithMessage(notifications, controller, cancellationToken);
-                return;
-            }
+                notifications.ProgressChanged(Strings.ConnectionStepValidatinCredentials);
+                await this.host.SonarQubeService.ConnectAsync(connection, cancellationToken);
 
-            if (connection.Organization == null &&
-                this.host.SonarQubeService.HasOrganizationsSupport(connection, cancellationToken))
-            {
-                notifications.ProgressChanged(Strings.ConnectionStepRetrievingOrganizations);
-                OrganizationInformation[] organizations;
-                if (!this.host.SonarQubeService.TryGetOrganizations(connection, cancellationToken, out organizations))
+                if (connection.Organization == null &&
+                    this.host.SonarQubeService.HasOrganizationsFeature)
                 {
-                    AbortWithMessage(notifications, controller, cancellationToken);
-                    return;
-                }
+                    notifications.ProgressChanged(Strings.ConnectionStepRetrievingOrganizations);
+                    var organizations = await this.host.SonarQubeService.GetAllOrganizationsAsync(cancellationToken);
 
-                if (organizations.Length <= 1)
-                {
-                    connection.Organization = null;
-                }
-                else
-                {
-                    connection.Organization = AskUserToSelectOrganizationOnUIThread(organizations);
-                    if (connection.Organization == null) // User clicked cancel
+                    if (organizations.Count <= 1) // Only 1 org means the SQ server is recent enough but the feature is not active
                     {
-                        AbortWithMessage(notifications, controller, cancellationToken);
-                        return;
+                        connection.Organization = null;
                     }
+                    else
+                    {
+                        connection.Organization = AskUserToSelectOrganizationOnUIThread(organizations);
+                        if (connection.Organization == null) // User clicked cancel
+                        {
+                            AbortWithMessage(notifications, controller, cancellationToken); // TODO: Might be worth throwing
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this 'TODO' comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS1135)